core/arch/arm/kernel/user_ta.c

```diff
@@ -645,11 +645,11 @@ static TEE_Result rpc_load(const TEE_UUID *uuid, struct shdr **ta,
 	if (res != TEE_SUCCESS)
 		return res;
 
-	thread_rpc_alloc_payload(params[1].u.tmem.size, &phta, &cta);
-	if (!phta)
+	*mobj = thread_rpc_alloc_payload(params[1].u.tmem.size, &cta);
+	if (!*mobj)
 		return TEE_ERROR_OUT_OF_MEMORY;
 
-	*ta = phys_to_virt(phta, MEM_AREA_NSEC_SHM);
+	*ta = mobj_get_va(*mobj, 0);
```

Comment:
This used to insure was SHM. Should you add `mobj_is_nonsec(*mobj);` or rely on `thread_rpc_alloc_payload()` to be safe? I guess the later is valid. So ok for this change. 
Then the case `(!*ta)` should either be asserted or panicked.